packages/app-desktop/gui/NoteEditor/utils/useFormNote.ts

```diff
@@ -65,7 +65,17 @@ export default function useFormNote(dependencies: HookDependencies) {
 	const [isNewNote, setIsNewNote] = useState(false);
 	const prevSyncStarted = usePrevious(syncStarted);
 	const previousNoteId = usePrevious(formNote.id);
-	const [resourceInfos, setResourceInfos] = useState<ResourceInfos>({});
+	const resourceInfos = useRef<ResourceInfos>({});
+
+	// Only when a new value is different from the old value,
+	// resourceInfos is updated. Note that resourceInfos does not often
+	// change when a note is switched, because it is often the case
+	// that a note's resourceInfos is {}.
+	const updateResourceInfos = (newResourceInfos: ResourceInfos) => {
+		if (resourceInfosChanged(resourceInfos.current, newResourceInfos)) {
```

Comment:
maybe it will be easier to search for the term `Infos` and replace it with `Info`.  ;-)

Reply:
That wouldn't work because you see a ResourceInfos is a map of ResourceInfo - https://github.com/laurent22/joplin/blob/0b01b5b0ef7a2e9ac9e96a53b85bdf44e14d2f18/packages/app-desktop/gui/NoteEditor/utils/types.ts#L149 So if we rename everything to Info that would cause all kind of problems.